### PR TITLE
Support custom CSV delimiters

### PR DESCRIPTION
**Pull request by blp**
TPC-H uses `|` instead of `,` and I'm sure that other datasets use other characters too.

### CODE REVIEW

File: crates/adapters/src/adhoc/table.rs [hunk 1]
```diff
@@ -49,6 +50,7 @@ pub const fn datafusion_arrow_serde_config() -> &'static SqlSerdeConfig {
         date_format: DateFormat::String("%Y-%m-%d"),
         decimal_format: DecimalFormat::String,
         variant_format: VariantFormat::JsonString,
+        delimiter: CsvDelimiter::DEFAULT,
```
Comment (ryzhyk) [inline crates/adapters/src/adhoc/table.rs:53]: It's odd that this needs to be part of `SqlSerdeConfig`, which is only supposed to contain properties related to deserializing individual data types, not how the parses handles the envelope. Is it possible to refactor this to not have `delimiter` in `SqlSerdeConfig`?

File: web-console/src/lib/services/manager/schemas.gen.ts [hunk 2]
```diff
@@ -495,21 +495,35 @@ snapshot.  The condition must be a valid SQL Boolean expression that can be used
...
 
-
 When this option is specified, and \`mode\` is one of \`snapshot\` or \`snapshot_and_follow\`,
-the snapshot of the table is ingested in the timestamp order.  This setting is required
-for tables declared with the
-[\`LATENESS\`](https://docs.feldera.com/sql/streaming#lateness-expressions) attribute
-in Feldera SQL. It impacts the performance of the connector, since data must be sorted
-before pushing it to the pipeline; therefore it is not recommended to use this
-settings for tables without \`LATENESS\`.`,
+table rows are ingested in the timestamp order, respecting the
```
Comment (mihaibudiu) [inline web-console/src/lib/services/manager/schemas.gen.ts:506]: I thought I saw this in a prior commit

**Review by mihaibudiu — APPROVED** [review]

**Review by ryzhyk — COMMENTED** [review]
Other comments:
* you probably want to update `docs/formats/csv.md` as well.
* another useful option is to enable/disable CSV headers, but doesn't have to be in this PR.

File: crates/adapters/src/adhoc/table.rs (hunk 1 above)
Comment (blp) [reply crates/adapters/src/adhoc/table.rs:53]: OK, I'll figure it out. I didn't understand what was happening with SqlSerdeConfig.

**Comment by blp** [comment]
> Other comments:
> 
>     * you probably want to update `docs/formats/csv.md` as well.

Updated.

>     * another useful option is to enable/disable CSV headers, but doesn't have to be in this PR.

I added this but it's a little weird, because the csv writer can't write headers in all the cases where it can write records. See "However, if `has_headers` were enabled..." just below the code example at https://docs.rs/csv/latest/csv/struct.Writer.html#structs. So, I didn't enable headers for csv output, only for input.

File: crates/adapters/src/adhoc/table.rs (hunk 1 above)
Comment (blp) [reply crates/adapters/src/adhoc/table.rs]: Fixed.

**Review by ryzhyk — APPROVED** [review]

**Comment by mihaibudiu** [comment]
Can I put this in the queue?

**Pull request merged**